### PR TITLE
Do not create String instances in 'Strings' methods accepting StringBuilder

### PR DESCRIPTION
**Pull request by sabi0**
Imagine you are doing something like this:
```java
StringBuilder sb = new StringBuilder();
// ... adding some heavy data to 'sb'
sb.append("indices[");
Strings.arrayToDelimitedString(indices, ",", sb);
sb.append("], ");
sb.append("types[");
Strings.arrayToDelimitedString(types, ",", sb);
sb.append("]");
```

And before you know it, you've just allocated two extra 5 MB `String` instances on the heap.

### CODE REVIEW

File: modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java [hunk 1]
```diff
@@ -90,12 +90,12 @@ public void shutdown() throws Exception {
     public void testCorsConfig() {
         final Set<String> methods = new HashSet<>(Arrays.asList("get", "options", "post"));
         final Set<String> headers = new HashSet<>(Arrays.asList("Content-Type", "Content-Length"));
-        final String suffix = randomBoolean() ? " " : ""; // sometimes have a leading whitespace between comma delimited elements
+        final String prefix = randomBoolean() ? " " : ""; // sometimes have a leading whitespace between comma delimited elements
```
Comment (rjernst) [inline modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java]: Can you please undo this unrelated change?

**Review by rjernst — CHANGES_REQUESTED** [review]

**Comment by rjernst** [comment]
@elasticmachine test this please

File: modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java (hunk 1 above)
Comment (sabi0) [inline modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java]: This value is passed as 'prefix' parameter to the `collectionToDelimitedString(coll, delim, prefix, suffix)`.
And since that method was affected by this change I thought it might be a good idea to also eliminate this possible source of confusion.

But sure, I'll undo it tomorrow.

Comment (javanna) [inline modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java]: I think it's a good change and not completely unrelated. Why undo it @rjernst ?

Comment (rjernst) [inline modules/transport-netty4/src/test/java/org/elasticsearch/http/netty4/Netty4HttpServerTransportTests.java]: I'm fine with it, given the explanation. I only saw that it seemed to be a variable name change that was unnecessary.

**Comment by rjernst** [comment]
@elasticmachine retest this please, just once, please please please

**Comment by sabi0** [comment]
The build was failing due to some unresolvable dependency. I didn't touch any Gradle files though.
Am I expected to do something about that? (rebase the branch?)

**Comment by rjernst** [comment]
@sabi0 We've had issues with our PR CI system.  I'll try it once more, otherwise I will run tests manually.

@elasticmachine test this please

**Comment by sabi0** [comment]
If your CI is back to normal would you mind pulling this in?

**Review by rjernst — APPROVED** [review]

**Comment by rjernst** [comment]
@sabi0 I'm sorry I lost track of this! I just tested and it is good. I will merge now.

**Comment by rjernst** [comment]
Thanks @sabi0!
